src/components/duelling-picklist/picklist.component.js

```diff
@@ -0,0 +1,87 @@
+import React, { useMemo, useCallback } from 'react';
+import PropTypes from 'prop-types';
+import { TransitionGroup } from 'react-transition-group';
+
+import Events from '../../utils/helpers/events';
+import { StyledPicklist, StyledEmptyContainer } from './duelling-picklist.style';
+
+export const Picklist = ({ disabled, children, renderPlaceholder }) => {
+  const isEmpty = useMemo(() => !React.Children.count(children), [children]);
+
+  const refs = useMemo(() => Array.from({
+    length: React.Children.count(children)
+  }, () => React.createRef()), [children]);
+
+  const focusItem = useCallback((ev, index) => {
+    ev.preventDefault();
+    if (index === -1) {
+      refs[refs.length - 1].current.focus();
+    } else if (index === refs.length) {
+      refs[0].current.focus();
+    } else {
+      refs[index].current.focus();
+    }
+  }, [refs]);
+
+  const handleKeyboardAccessibility = useCallback((ev, index) => {
+    if (Events.isUpKey(ev)) {
+      focusItem(ev, index - 1);
+    }
+    if (Events.isDownKey(ev)) {
+      focusItem(ev, index + 1);
+    }
+    if (Events.isHomeKey(ev)) {
+      focusItem(ev, 0);
+    }
+    if (Events.isEndKey(ev)) {
+      focusItem(ev, refs.length - 1);
+    }
+  }, [focusItem, refs]);
+
+  const content = React.Children.map(children, (child, index) => React.cloneElement(child, {
+    ref: refs[index],
+    index,
+    disabled,
+    handleKeyboardAccessibility
+  }));
+
+  return (
+    <StyledPicklist>
+      {isEmpty && (
+        <StyledEmptyContainer>
+          {renderPlaceholder}
+        </StyledEmptyContainer>
+      )}
+      <TransitionGroup component={ null }>
+        {content}
+      </TransitionGroup>
+    </StyledPicklist>
+  );
+};
+
+Picklist.propTypes = {
+  /** List of PicklistItem elements */
+  children: PropTypes.node,
+  /** Placeholder to be rendered when list is empty */
+  renderPlaceholder: PropTypes.node,
+  /** Indicate if component is disabled */
+  disabled: PropTypes.bool
+};
+
+export const areEqual = (prevProps, nextProps) => {
```

Comment:
Is this going be to used anywhere other than in the for the `useMemo` optimisation and in the tests? Just not sure it should be exposed as part of the component api 🤔

Reply:
Yeah unfortunately that's the only way that I've found to test this method :/

Reply:
Yeah I suppose as testing libraries get better at handling hooks this can be looked at, but I'm cool with it for now